eZ/Publish/Core/FieldType/DateAndTime/SearchField.php

```diff
@@ -28,10 +29,17 @@ class SearchField implements Indexable
      */
     public function getIndexData(Field $field, FieldDefinition $fieldDefinition)
     {
+        if (isset($field->value->data['timestamp']) && $field->value->data['timestamp'] !== '') {
```

Comment:
`if ($field->value->data !== null)` is enough because DateAndTime Converter [ensures the same thing as above](https://github.com/ezsystems/ezpublish-kernel/blob/v6.7.3/eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/DateAndTimeConverter.php#L57-L68).